### PR TITLE
feat(primitives): support inert prop

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Description of changes

Adds support for the HTML `inert` attribute: https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/inert

React elements do not support the prop by default, so adding it in to make it work like any other HTML attribute. 

#### Issue #, if available

#2632

#### Description of how you validated changes

Added unit test and also tried it on the docs site:
CleanShot 2023-01-06 at 13 22 42@2x

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/src/primitives/types/base.ts [hunk 1]
```diff
@@ -19,6 +19,13 @@ export interface BaseComponentProps {
    * Used to provide a `data-testid` attribute for testing purposes
    */
   testId?: string;
+
+  /**
+   * @description
+   * Support for the HTML `inert` attribute
+   * https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/inert
+   */
+  inert?: string | null;
```
Comment (zchenwei) [inline packages/react/src/primitives/types/base.ts:28]: Should be boolean?

Comment (zchenwei) [reply packages/react/src/primitives/types/base.ts:28]: Then you can also `inert: inert ? '' : null,`

Comment (hbuchel) [reply packages/react/src/primitives/types/base.ts:28]: Yeah I'm wondering if the type as `string | null` might be confusing to developers who aren't familiar with its implementation (i.e. that gh issue I linked in the feature request)?

Comment (dbanksdesign) [reply packages/react/src/primitives/types/base.ts]: yea that makes sense, updated to boolean

File: packages/react/src/primitives/View/View.tsx [hunk 2]
```diff
@@ -28,6 +29,7 @@ const ViewPrimitive = <Element extends ElementType = 'div'>(
       'data-testid': testId,
       disabled: isDisabled,
       ref,
+      inert: inert ? '' : null,
```
Comment (0618) [inline packages/react/src/primitives/View/View.tsx:32]: What about this?

```suggestion
           ...(inert && { inert: "" }),
```

Comment (0618) [reply packages/react/src/primitives/View/View.tsx:32]: also, if the type of `inert` is `boolean`, should it be boolean here?

`      ...(true && { inert: true }),`

Comment (zchenwei) [reply packages/react/src/primitives/View/View.tsx:32]: I guess if you use boolean value here React will render `inert=true/false` but we just want `inert` on the element

**Review by zchenwei — APPROVED** [review]

**Review by hbuchel — APPROVED** [review]

File: packages/react/src/primitives/View/View.tsx (hunk 2 above)
Comment (dbanksdesign) [reply packages/react/src/primitives/View/View.tsx:32]: This is actually a React issue, we can't pass the boolean `true` here or it won't actually get rendered in the DOM. See this quick example:

https://codesandbox.io/s/crazy-wu-02mz2p?file=/src/App.js

Comment (dbanksdesign) [reply packages/react/src/primitives/View/View.tsx:32]: Heather also documented this in the issue (Thanks Heather!) https://github.com/aws-amplify/amplify-ui/issues/2632

**Pull request merged**